# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/dietlibc/pkg_patch/pkg_textutils.patch-x
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./lib/getline.c.orig	Wed May  1 12:32:27 2002
+++ ./lib/getline.c	Wed May  1 12:35:45 2002
@@ -22,10 +22,8 @@
 # include <config.h>
 #endif
 
-/* The `getdelim' function is only declared if the following symbol
-   is defined.  */
-#ifndef _GNU_SOURCE
-# define _GNU_SOURCE 1
+#ifdef _GNU_SOURCE
+# undef _GNU_SOURCE
 #endif
 
 #include <stdio.h>
--- ./lib/getline.h.orig	Thu May 13 22:51:21 1999
+++ ./lib/getline.h	Wed May  1 12:35:45 2002
@@ -17,6 +17,10 @@
 #ifndef GETLINE_H_
 # define GETLINE_H_ 1
 
+#ifdef _GNU_SOURCE
+# undef _GNU_SOURCE
+#endif
+
 # include <stdio.h>
 
 # ifndef PARAMS
--- ./src/paste.c.orig	Wed May  1 12:40:33 2002
+++ ./src/paste.c	Wed May  1 12:40:47 2002
@@ -54,11 +54,11 @@
 /* Indicates that no delimiter should be added in the current position. */
 #define EMPTY_DELIM '\0'
 
-static FILE dummy_closed;
+static int dummy_closed;
 /* Element marking a file that has reached EOF and been closed. */
 #define CLOSED (&dummy_closed)
 
-static FILE dummy_endlist;
+static int dummy_endlist;
 /* Element marking end of list of open files. */
 #define ENDLIST (&dummy_endlist)
 
--- ./config.hin.orig	2002-10-09 12:42:20.000000000 +0200
+++ ./config.hin	2002-10-09 12:42:28.000000000 +0200
@@ -1269,11 +1269,6 @@
 /* Number of bits in a file offset, on hosts where this is settable. */
 #undef _FILE_OFFSET_BITS
 
-/* Enable GNU extensions on systems that have them.  */
-#ifndef _GNU_SOURCE
-# undef _GNU_SOURCE
-#endif
-
 /* Define to 1 to make fseeko visible on some hosts (e.g. glibc 2.2). */
 #undef _LARGEFILE_SOURCE
 
--- ./src/sys2.h.orig	2002-10-09 13:03:20.000000000 +0200
+++ ./src/sys2.h	2002-10-09 13:06:06.000000000 +0200
@@ -264,6 +264,14 @@
 # include <inttypes.h> /* for the definition of UINTMAX_MAX */
 #endif
 
+#ifdef __CHAR_UNSIGNED__
+#   define CHAR_MIN     0
+#   define CHAR_MAX     UCHAR_MAX
+#else
+#   define CHAR_MIN     SCHAR_MIN
+#   define CHAR_MAX     SCHAR_MAX
+#endif
+
 #include <ctype.h>
 
 /* Jim Meyering writes:
